src/frontend/src/optimizer/plan_node/logical_source.rs

```diff
@@ -197,6 +241,73 @@ impl LogicalSource {
             kafka_timestamp_range: range,
         }
     }
+
+    #[must_use]
+    fn rewrite_to_stream_batch_source(&self) -> Self {
+        let column_catalog = self.core.column_catalog.clone();
+        // Filter out the generated columns.
+        let row_id_index = Self::rewrite_row_id_idx(&column_catalog, self.core.row_id_index);
+        let source_column_catalogs = column_catalog
+            .into_iter()
+            .filter(|c| !c.is_generated())
+            .collect_vec();
+        let core = generic::Source {
+            catalog: self.core.catalog.clone(),
+            column_catalog: source_column_catalogs,
+            pk_col_ids: self.core.pk_col_ids.clone(),
+            row_id_index,
```

Comment:
Columns in Stream/Batch source node indicate the actual columns it will produce on source node instead of the schema on logical source. Will add comments